src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -48,6 +48,13 @@ export interface EslintStrategy {
 	/** After applying target patterns, last chance to filter any unsupported files */
 	filterUnsupportedPaths(paths: string[]): string[];
 
+	/** Filters out any rules that should be excluded from the catalog */
+	filterDisallowedRules(rulesByName: Map<string,ESRule>): Map<string,ESRule>;
+
+	ruleDefaultEnabled(name: string): boolean;
+
+	getDefaultConfig(ruleName: string): LooseObject;
```

Comment:
Could this be a strongly typed class instead of LooseObject? The code that interacts with this seems like its more complicated than it needs to be since it has to be decoded.

Reply:
I'm not sure it can. If anything, I think in the long term we'll probably need to broaden it to `any`. Some of the LWC rules use objects, but some of the base ESLint rules (e.g. `quotes`) accept a string.

Reply:
Could we extend the LooseObject or wrap it in a class that encapsulates the complicated if/else cases? This seems easier to test and removes if from the logic of the Engine.